src/test/regress/sql/partitioning_issue_3970.sql

```diff
@@ -0,0 +1,48 @@
+-- test cases for #3970
+SET citus.next_shard_id TO 1690000;
+SET citus.shard_count TO 4;
+SET citus.shard_replication_factor TO 1;
+
+CREATE SCHEMA test_3970;
+SET search_path = test_3970;
+
+--1. create a partitioned table
+CREATE TABLE part_table (
+    work_ymdt timestamp without time zone NOT NULL,
+    seq bigint NOT NULL,
+    my_seq bigint NOT NULL,
+    work_memo character varying(150),
+    CONSTRAINT work_memo_check CHECK ((octet_length((work_memo)::text) <= 150))
+)
+PARTITION BY RANGE (work_ymdt);
+
+--2. perform create_distributed_table
+SELECT create_distributed_table('part_table', 'seq');
+
+--3. add a partition
+CREATE TABLE part_table_p202008 PARTITION OF part_table FOR VALUES FROM ('2020-08-01 00:00:00') TO ('2020-09-01 00:00:00');
+
+--4. add a check constraint
+ALTER TABLE part_table ADD CONSTRAINT my_seq CHECK (my_seq > 0);
+
+--5. add a partition
+CREATE TABLE part_table_p202009 PARTITION OF part_table FOR VALUES FROM ('2020-09-01 00:00:00') TO ('2020-10-01 00:00:00');
+
+-- check the constraint names on the coordinator node
+\d+ test_3970.part_table*
```

Comment:
These outputs are very verbose and hard to spot quickly what it wants to verify.

Can we query pg_constraint or such to get only the relevant info?

Reply:
I dropped `\d+` lines and wrote queries using catalogue tables.

https://github.com/citusdata/citus/blob/004907555e2906fc4cdffb25c0d966b53d11d551/src/test/regress/expected/partitioning_issue_3970.out#L48-L61